go/host/host.go

```diff
@@ -408,37 +397,11 @@ func (h *host) startProcessing() { //nolint:gocognit
 	// - Process new Transactions gossiped from L2 Peers
 	for {
 		select {
-		case blockAndParent := <-h.l1BlockRPCCh:
-			roundInterrupt = triggerInterrupt(roundInterrupt)
-			err := h.processL1Block(blockAndParent.parent, false)
-			if err != nil {
-				var rejErr *common.BlockRejectError
-				if errors.As(err, &rejErr) {
-					// this is a common error (the parent has usually already been processed and is being sent just in case)
-					h.logger.Debug("Rejected parent block.", log.ErrKey, rejErr, "enclaveHead", rejErr.L1Head)
-				} else {
-					h.logger.Info("Could not process parent block. ", log.ErrKey, err)
-				}
-			}
-			err = h.processL1Block(blockAndParent.block, true)
-			if err != nil {
-				var rejErr *common.BlockRejectError
-				if errors.As(err, &rejErr) {
-					h.logger.Info("Rejected latest block.", log.ErrKey, rejErr, "enclaveHead", rejErr.L1Head)
-				} else {
-					h.logger.Warn("Could not process latest block.", log.ErrKey, err)
-				}
-			}
-
-		case f := <-h.forkRPCCh:
+		case b := <-blockStream:
 			roundInterrupt = triggerInterrupt(roundInterrupt)
-			for i, blk := range f {
-				isLatest := i == (len(f) - 1)
-				err := h.processL1Block(blk, isLatest)
-				if err != nil && isLatest {
-					h.logger.Warn("Could not process latest fork block received via RPC.", log.ErrKey, err)
-				}
-			}
+			err := h.processL1Block(b, h.l1BlockProvider.IsLive(b.Hash()))
+			// handle the error, replace the blockStream if necessary (e.g. if stream needs resetting based on enclave's reported L1 head)
+			blockStream = h.handleProcessBlockErr(b, blockStream, err)
```

Comment:
this is a bit unusual.
Would it be possible to call this function only if `err != nil`?

Reply:
Yes whoops, that's much better